motionplan/planManager.go

```diff
@@ -508,13 +494,59 @@ func (pm *planManager) plannerSetupFromMoveRequest(
 		}
 	}
 
+	// extract inputs corresponding to the frame
+	frameInputs, err := pm.frame.mapToSlice(seedMap)
+	if err != nil {
+		return nil, err
+	}
+
+	// create robot collision entities
+	movingGeometriesInFrame, err := pm.frame.Geometries(frameInputs)
+	movingRobotGeometries := movingGeometriesInFrame.Geometries() // solver frame returns geoms in frame World
+	if err != nil {
+		if len(movingRobotGeometries) == 0 {
+			return nil, err // no geometries defined for frame
+		}
+	}
+	if pm.useTPspace {
```

Comment:
I'm having a hard time following this case can you explain why we need it?

Reply:
The answer to that is also the answer to your above question regarding moving pieces out of `createAllCollisionConstraints`.

If we are starting a ptg plan at a different place than the origin, then that translation must be represented in the geometries, all of which need to be in the "correct" position when transformed to the world frame.

At this point in the plan manager, a ptg plan is moving to a goal in the world frame, and the start pose is the base's location relative to world. Since nothing providing a geometry knows anything about where the base is relative to world, any geometries need to be transformed by the start position to place them correctly in world. Thus `startGeoms = append(startGeoms, geometry.Transform(from))`.

As for why this was taken out of `createAllCollisionConstraints`; the options were to do that, or to pass the start pose and the useTPspace bool into `createAllCollisionConstraints`. Thinking about what `createAllCollisionConstraints` actually does and how that area of code is organized, it makes more sense to have it operate on specific arrays of geometries. Business logic like this fits better in the caller.

Reply:
Thanks this makes sense.  Can you add a short comment here explaining this?